Detectors/GlobalTracking/src/MatchHMP.cxx

```diff
@@ -458,11 +493,12 @@ void MatchHMP::doMatching()
           hmpTrk = nullptr;
           delete hmpTrkConstrained;
           hmpTrkConstrained = nullptr;
+          delete matching;
+          matching = nullptr;
+          oneEventClusters.clear();
           continue;
```

Comment:
There is no reason to use new/delete in these loops, instead of using `new` at 
https://github.com/AliceO2Group/AliceO2/blob/5f137b7119bc80e76ce18ef4d037535c6bc8e90a/Detectors/GlobalTracking/src/MatchHMP.cxx#L418
https://github.com/AliceO2Group/AliceO2/blob/5f137b7119bc80e76ce18ef4d037535c6bc8e90a/Detectors/GlobalTracking/src/MatchHMP.cxx#L426
https://github.com/AliceO2Group/AliceO2/blob/5f137b7119bc80e76ce18ef4d037535c6bc8e90a/Detectors/GlobalTracking/src/MatchHMP.cxx#L534 
and following deletes, just use stack objects (no need to delete them)

Reply:
so for "matching" was my first choice but then since I pass it to a method  of an other class that modifies it, does not work, I need to use the pointer! If I do just like this "MatchInfo *matching" it compiles but crashes in execution.
Then I need to create copies of the tracks (trackHMP and hmpTrkConstrained), can I do without using "= new ...."?

Reply:
For the `matching`: you can simply use at https://github.com/AliceO2Group/AliceO2/blob/5f137b7119bc80e76ce18ef4d037535c6bc8e90a/Detectors/GlobalTracking/src/MatchHMP.cxx#L418
```
MatchInfo (999999, mTrackGid[type][cacheTrk[itrk]]);
```
then change all `matching->` by `matching.`, all `*matching` by `matching` and passing the pointer to `&matching`.
And remove all the lines like https://github.com/AliceO2Group/AliceO2/blob/5f137b7119bc80e76ce18ef4d037535c6bc8e90a/Detectors/GlobalTracking/src/MatchHMP.cxx#L437-L438

Idem for https://github.com/AliceO2Group/AliceO2/blob/5f137b7119bc80e76ce18ef4d037535c6bc8e90a/Detectors/GlobalTracking/src/MatchHMP.cxx#L426 and https://github.com/AliceO2Group/AliceO2/blob/5f137b7119bc80e76ce18ef4d037535c6bc8e90a/Detectors/GlobalTracking/src/MatchHMP.cxx#L534 (and you don't need to define the `hmpTrkConstrained` pointer in advance).